oar/core/advisory.py

```diff
@@ -629,3 +661,99 @@ def has_blocking_secruity_alert(self):
             return blocking
         else:
             return False
+        
+    def get_unhealthy_builds(self):
+        """
+        Get unhealthy builds of advisory.
+
+        Returns:
+            list: Unhealthy builds.
+        """
+        nvrs = self.get_build_nvrs()
+
+        unhealthy_builds = []
+
+        for n in nvrs:
+            build_grades = self.get_build_grades(n)
+            for bg in build_grades:
+                if not util.is_grade_healthy(bg["grade"]):
+                    unhealthy_builds.append(bg)
+
+        return unhealthy_builds
+        
+    def get_build_grades(self, nvr):
+        """
+        Get all architecture grades of build.
+
+        Returns:
+            list: All architecture grades of build {nvr, architecture, grade}.
+        """
+        nvr_url = "https://pyxis.engineering.redhat.com/v1/images/nvr/%s?include=data.freshness_grades&include=data.architecture" % nvr
+        resp = requests.get(nvr_url, auth=self._auth, verify=self.ssl_verify)
+        nvr_grades = []
+
+        if resp.ok:
+            for arch in resp.json()["data"]:
+                effective_grade = None
+                for fg in arch["freshness_grades"]:                    
+                    checked_grade = fg["grade"]
+                    start_date = parser.parse(fg["start_date"])
+
+                    # skip if checked grade is not effective yet
+                    if start_date > datetime.now(timezone.utc):
+                        continue
+
+                    # skip if checked grade is no longer effective
+                    if "end_date" in fg and parser.parse(fg["end_date"]) < datetime.now(timezone.utc):
+                        continue
+
+                    # skip if checked grade is smaller than current effective grade (if intervals are overlapping, bigger grade is taken)
+                    if effective_grade and checked_grade < effective_grade:
+                        continue
+
+                    # save new effective grade
+                    effective_grade = checked_grade
+
+                nvr_grades.append({"nvr": nvr, "arch": arch["architecture"], "grade": effective_grade})
+        else:
+            raise AdvisoryException(f"error when accessing build nvr - {nvr}")
+        return nvr_grades
+        
+    def get_build_nvrs(self):
```

Comment:
I think you have to impl this func to get builds, because `Advisory` is inherited from `Erratum`, `self.errata_builds` will be filled when the object is initialized.
xref: https://github.com/red-hat-storage/errata-tool/blob/25489e09945fccb29e5bc6e5de3208bca097d0b0/errata_tool/erratum.py#L492

Reply:
I will take a look. Thanks.

Reply:
Removed. Thanks, I missed that the method exists.